sympy/core/function.py

```diff
@@ -855,57 +860,32 @@ class Derivative(Expr):
 
     Derivative wrt non-Symbols:
 
-    This class also allows derivatives wrt non-Symbols that have _diff_wrt
+    This class don't allows derivatives wrt non-Symbols that have _diff_wrt
     set to True, such as Function and Derivative. When a derivative wrt a non-
     Symbol is attempted, the non-Symbol is temporarily converted to a Symbol
     while the differentiation is performed.
 
-    Note that this may seem strange, that Derivative allows things like
-    f(g(x)).diff(g(x)), or even f(cos(x)).diff(cos(x)).  The motivation for
```

Comment:
I think you should restore this paragraph, with just the "or even f(cos(x)).diff(cos(x))" removed, and "Function and Derivative objects" below replaced with "undefined Function and derivatives of undefined functions".